contracts/governance/InterchainGovernance.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { AxelarExecutable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol';
+import { TimeLock } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/TimeLock.sol';
+import { IInterchainGovernance } from '../interfaces/IInterchainGovernance.sol';
+
+contract InterchainGovernance is AxelarExecutable, TimeLock, IInterchainGovernance {
+    enum GovernanceCommand {
+        ScheduleTimeLockProposal,
+        CancelTimeLockProposal
+    }
+
+    bytes32 public immutable governanceChainHash;
+    bytes32 public immutable governanceAddressHash;
+
+    // solhint-disable-next-line no-empty-blocks
+    constructor(
+        address gatewayAddress,
+        string memory governanceChain_,
+        string memory governanceAddress_,
+        uint256 minimumTimeDelay
+    ) AxelarExecutable(gatewayAddress) TimeLock(minimumTimeDelay) {
+        governanceChainHash = keccak256(bytes(governanceChain_));
+        governanceAddressHash = keccak256(bytes(governanceAddress_));
+    }
+
+    function executeProposal(address target, bytes calldata callData) external payable {
+        bytes32 proposalHash = keccak256(abi.encodePacked(target, callData, msg.value));
+
+        _executeTimeLock(proposalHash);
+
+        (bool success, ) = target.call{ value: msg.value }(callData);
+
+        if (!success) {
+            revert ExecutionFailed();
+        }
+
+        emit ProposalExecuted(proposalHash);
+    }
+
+    function _execute(
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
```

Comment:
Probably not a super big deal, but contracts that inherit this module will not be able to encode the payload on-chain as solidity doesn't support implicit conversion from bytes memory to bytes calldata.

Reply:
The `_execute` method is always called by a crosschain GMP call and the payload comes from the external call.